tests/auditd/test_auditd.py

```diff
@@ -14,64 +13,88 @@
 ]
 
 RULES_DIR = "/etc/audit/rules.d/"
+NSENTER_CMD = "nsenter --target 1 --pid --mount --uts --ipc --net"
 DOCKER_EXEC_CMD = "docker exec {} bash -c "
-NSENTER_CMD = "nsenter --target 1 --pid --mount --uts --ipc --net "
+AUDITD_CMD = DOCKER_EXEC_CMD.format("auditd") + "'{} {}'"
+AUDITD_WATCHDOG_CMD = DOCKER_EXEC_CMD.format("auditd_watchdog") + "'{} {}'"
 CURL_HTTP_CODE_CMD = "curl -s -o /dev/null -w \%\{http_code\} http://localhost:50058"   # noqa: W605
 CURL_CMD = "curl http://localhost:50058"    # noqa: W605
 logger = logging.getLogger(__name__)
 
 
-def verify_container_running(duthost, container_name):
-    is_running = is_container_running(duthost, container_name)
-    if not is_running:
-        pytest.skip("Container {} is not running".format(container_name))
+def is_log_valid(pattern, logs):
+    for log in logs:
+        if pattern in log and "ansible-ansible" not in log:
+            return True
+    return False
+
+
+def extract_audit_timestamp(logs):
+    timestamp_regex = r'audit\((\d+\.\d+):\d+\)'
+    for log in logs:
+        if "ansible-ansible" not in log:
+            match = re.search(timestamp_regex, log)
+            if match:
+                return match.group(1)
+    return ''
+
 
+def extract_audit_timestamp_with_seq(logs):
```

Comment:
>extract_audit_timestamp_with_seq [](http://example.com/codeflow?start=4&length=32)

It is quite similar to above, possible to consolidate?

Reply:
fixed